src/test/kotlin/no/nav/tiltaksarrangor/service/TilgangServiceTest.kt

```diff
@@ -0,0 +1,173 @@
+package no.nav.tiltaksarrangor.service
+
+import io.kotest.matchers.shouldBe
+import io.kotest.matchers.shouldNotBe
+import io.mockk.clearMocks
+import io.mockk.coEvery
+import io.mockk.mockk
+import no.nav.tiltaksarrangor.IntegrationTest
+import no.nav.tiltaksarrangor.client.amtarrangor.AmtArrangorClient
+import no.nav.tiltaksarrangor.ingest.model.AnsattDto
+import no.nav.tiltaksarrangor.ingest.model.AnsattPersonaliaDto
+import no.nav.tiltaksarrangor.ingest.model.AnsattRolle
+import no.nav.tiltaksarrangor.ingest.model.NavnDto
+import no.nav.tiltaksarrangor.ingest.model.TilknyttetArrangorDto
+import no.nav.tiltaksarrangor.ingest.model.VeilederDto
+import no.nav.tiltaksarrangor.ingest.model.Veiledertype
+import no.nav.tiltaksarrangor.ingest.model.toAnsattDbo
+import no.nav.tiltaksarrangor.model.exceptions.UnauthorizedException
+import no.nav.tiltaksarrangor.repositories.AnsattRepository
+import no.nav.tiltaksarrangor.testutils.DbTestDataUtils
+import no.nav.tiltaksarrangor.testutils.DbTestDataUtils.shouldBeCloseTo
+import no.nav.tiltaksarrangor.testutils.SingletonPostgresContainer
+import no.nav.tiltaksarrangor.utils.sqlParameters
+import org.junit.jupiter.api.AfterEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate
+import java.time.LocalDateTime
+import java.util.UUID
+
+class TilgangServiceTest : IntegrationTest() {
+	private val amtArrangorClient = mockk<AmtArrangorClient>()
```

Comment:
Hvorfor er denne en integrasjonstest?

Det er kanskje bare min preferanse, men jeg har altid mislikt blanding av mocking og integrasjoner slik. Men kanskje det fordelere om det er sånn mot at det er renere unit-tester.

Reply:
Ah, det har du rett i, det trenger den ikke å være :) Jeg trodde den måtte det for at jeg skulle kunne teste med en ordentlig database. I denne testen var jeg ikke interessert i å teste klienten, jeg var mest interessert i logikken i TilgangService og hva som faktisk blir lagret i databasen, og da gir testen mer verdi hvis jeg tester mot en faktisk database og ikke mot en mocket database.